### ISSUE TITLE
CS0649 is triggered for public serialized field of internal types.

### ISSUE THREAD

**Issue by jbevain** [microsoft/Microsoft.Unity.Analyzers#61]
#### Bug description
CS0649 is triggered for public serialized field of internal types.

#### To Reproduce
```csharp
internal class CameraScript : MonoBehaviour
{
    public Transform MyTransform;
}
```

#### Expected behavior
We should suppress the warning.

**Comment by aybe** [comment]
I also get that on the following:

```
    public sealed class TrackComponent : MonoBehaviour
    {
        [SerializeField]
        [CanBeNull]
        private Track _track; // <------------------- here

        [CanBeNull]
        public ITrack Track => _track;
    }
```

**Comment by aybe** [comment]
Another example:

```
    class MyClass : MonoBehaviour
    {
        public bool B; // CS0649

        private void Reset()
        {
            B = false;
        }

        private void Update()
        {
            if (B)
            {

            }
        }
    }
```

Now the thing is, no matter if you ever assign something to it, VS seems stuck on an old snapshot of its analysis -> even when you satisfy VS the error does not go away.

**Comment by aybe** [comment]
Also, in the previous example you will want to note that, other fields in class with exact same definition and usage are correctly seen by VS, i.e. no CS0649.

**Comment by sailro** [comment]
So I double checked, everything is now properly suppressed, thanks to https://github.com/microsoft/Microsoft.Unity.Analyzers/pull/73

Closing.

Thanks!
